autopush/senderids.py

```diff
@@ -78,17 +106,31 @@ def senderIDs(self):
             self._refresh()
         return self._senderIDs
 
+    def get(self, id):
+        """Return the associated record for a given SenderID"""
+        record = self._senderIDs.get(id)
+        if record:
+            record["senderID"] = id
+        return record
+
     def getID(self):
         """Return a randomly selected SenderID, refreshing if required"""
         self._refresh()
```

Comment:
Since getID is called in the GCM register function, which is called in a thread-pool, the refresh operation itself should be mutex locked and/or cached so that only a single refresh executes at a time.